plugin-server/src/worker/ingestion/event-pipeline/runner.ts

```diff
@@ -254,6 +255,8 @@ export class EventPipelineRunner {
         }
         const [normalizedEvent, timestamp] = normalizeResult.value
 
+        await internalHandlersStep(normalizedEvent)
```

Comment:
IMO, this is overgeneralized and the current implementation gives the impression that this is a well-supported feature/pattern, while it doesn't have any monitoring and lacks support for a lot of common features in ingestion. 

I would prefer if the survey logic was just a step like `prepareEventStep` and others. 

I'm in the middle of a large refactor, which will change the way it gets run, but for now this is ok and I'll refactor the surveys logic once I have all the pieces in place.